### PR TITLE
[SPARK-24937][SQL] Datasource partition table should load empty static partitions

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

How to reproduce:
```sql
spark-sql> CREATE TABLE tbl AS SELECT 1;
spark-sql> CREATE TABLE tbl1 (c1 BIGINT, day STRING, hour STRING)
         > USING parquet
         > PARTITIONED BY (day, hour);
spark-sql> INSERT INTO TABLE tbl1 PARTITION (day = '2018-07-25', hour='01') SELECT * FROM tbl where 1=0;
spark-sql> SHOW PARTITIONS tbl1;
spark-sql> CREATE TABLE tbl2 (c1 BIGINT)
         > PARTITIONED BY (day STRING, hour STRING);
spark-sql> INSERT INTO TABLE tbl2 PARTITION (day = '2018-07-25', hour='01') SELECT * FROM tbl where 1=0;
spark-sql> SHOW PARTITIONS tbl2;
day=2018-07-25/hour=01
spark-sql> 
```
1. Users will be confused about whether the partition data of `tbl1` is generated.
2. Inconsistent with Hive table behavior.

This pr fix this issues.

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
cc @gatorsmile @gengliangwang

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala [hunk 1]
```diff
@@ -284,6 +284,11 @@ object PartitioningUtils {
     }.mkString("/")
   }
 
+  def getPathFragment(partitions: TablePartitionSpec): String = partitions.map {
+    case (k, v) =>
+      escapePathName(k) + "=" + escapePathName(v)
+  }.mkString("/")
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala]: TablePartitionSpec is a map. The output of this func might be different from the actual schema, right?

For example, the partition schema is [partCol1, partCol2]. Your output could place partCol2 before partCol1

Comment (wangyum) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala]: You are right. Fixed it.

**Comment by gatorsmile** [comment]
@wangyum How about our Hive tables? Add a test case for Hive tables? They are created by CREATE TABLE instead of CREATE TABLE USING. You can use csv/json format in the test cases.

**Comment by wangyum** [comment]
retest this please.

**Comment by gengliangwang** [comment]
retest this please.

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master.

**Pull request closed without merging**